src/main/java/fi/otavanopisto/kuntaapi/server/controllers/NewsController.java

```diff
@@ -90,6 +111,32 @@ public AttachmentData getNewsArticleImageData(OrganizationId organizationId, New
     return entityController.sortEntitiesInNaturalOrder(result);
   }
   
+  private List<NewsArticle> searchNewsArticlesByTag(OrganizationId organizationId, String tag, Integer firstResult, Integer maxResults) { 
+    OrganizationId kuntaApiOrganizationId = idController.translateOrganizationId(organizationId, KuntaApiConsts.IDENTIFIER_NAME);
+    if (kuntaApiOrganizationId == null) {
+      logger.severe(String.format("Failed to translate organization %s into Kunta API id", organizationId.toString()));
+      return Collections.emptyList();
+    }
+    
+    SearchResult<NewsArticleId> searchResult = newsArticleSearcher.searchNewsArticlesByTag(kuntaApiOrganizationId.getId(), tag, firstResult != null ? firstResult.longValue() : null, maxResults != null ? maxResults.longValue() : null);
+    if (searchResult != null) {
+      List<NewsArticleId> newsArticleIds = searchResult.getResult();
+
+      List<NewsArticle> result = new ArrayList<>(newsArticleIds.size());
+
+      for (NewsArticleId newsArticleId : newsArticleIds) {
+        NewsArticle newsArticle = findNewsArticle(kuntaApiOrganizationId, newsArticleId);
+        if (newsArticle != null) {
+          result.add(newsArticle);
+        }
+      }
+      
+      return result;
+    }
+    
+    return null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)